backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/query/impl/ElasticsearchSearchQuery.java

```diff
@@ -77,11 +77,7 @@ public String toString() {
 
 	@Override
 	public long executeCount() {
-		ElasticsearchWork<SearchResult<T>> work = workFactory.search(
-				indexNames, routingKeys,
-				payload, searchResultExtractor,
-				firstResultIndex, 0L );
-		SearchResult<T> executeNoHits = queryOrchestrator.submit( work ).join();
-		return executeNoHits.getHitCount();
+		ElasticsearchWork<Long> work = workFactory.count( indexNames, routingKeys, payload );
```

Comment:
> I suspect that, if you added a test that retrieves the hit count from a projection query, it would fail: Elasticsearch would consider the payload invalid.

I think you're right. I guess the same.

Reply:
Good idea to add a backend IT for it.

Reply:
You're definitely right. 

> 11:50:58,502 (Hibernate Search: Elasticsearch transport thread-2) TRACE request:209 - HSEARCH400093: Executed Elasticsearch HTTP POST request to path '/indexname/_count' with query parameters {} in 7ms. Response had status 400 'Bad Request'. Request body: <
{
  "query": {
    "match_all": {}
  },
  "_source": [
    "string"
  ]
}
>. Response body: <
{
  "error": {
    "root_cause": [
      {
        "type": "parsing_exception",
        "reason": "request does not support [_source]",
        "line": 1,
        "col": 27
      }
    ],
    "type": "parsing_exception",
    "reason": "request does not support [_source]",
    "line": 1,
    "col": 27
  },
  "status": 400
}
>

Reply:
Fixed up in the factory method